File: restaurant-app-server/src/main/java/com/example/restaurantappserver/api/dto/IngredientDto.java
package com.example.restaurantappserver.api.dto;

import com.example.restaurantappserver.entity.Ingredient;
import com.example.restaurantappserver.entity.Item;
import com.example.restaurantappserver.entity.Recipe;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class IngredientDto {
    private Long id;
    private Long recipeId;
    private Double quantityNeeded;
    private Long itemId;
    private String itemName;
    private Double price;
    private Boolean isAvailable;
}

File: restaurant-app-client/src/app/models/ingredientDto.ts
export class Ingredient{
    "id" : number;
    "recipeId" : number;
    "quantityNeeded" : number;
    "itemId" : number;
    "itemName" : string;
    "price" : number;
    "isAvailable" : boolean;
    "isEdit" : boolean = false;
    "isNew" : boolean = false;
}